### ISSUE TITLE
Correct .Call() function signatures

### ISSUE THREAD

**Issue by joshuaulrich** [joshuaulrich/xts#317]
All C functions called via `.Call()` must return `SEXP` and have all `SEXP` arguments. This is not true for `do_merge_xts()` and `isXts()`. This may change the API, and impact RcppXts. Need to confirm.

Via email from @kalibera:

> Could you please check the signatures of foreign functions in xts? It
> seems that do_merge_xts is taking a non-SEXP argument, but it is called
> via .Call interface (all arguments have to be SEXP). Also isXts() is
> registered as a .Call foreign function, but it does not return SEXP.
> Both of these issues could probably lead to segfaults, so it would be
> great if you could fix them soon.

**Comment by joshuaulrich** [comment]
@eddelbuettel My preliminary checks suggest this is going to affect RcppXts. My understanding is that I need to do one of two things to `do_merge_xts()` and `isXts()`:

1. Do not register them as C-callable functions in `init.c`, or
2. Change the definitions so both functions only take `SEXP` arguments and return `SEXP` (`isXts()` returns `int`).

What are your thoughts/preferences?

**Comment by kalibera** [comment]
To clarify, not registering a function will not solve the problem. If a function is not registered, but is called via `.Call`, it still has to return `SEXP` and all of its arguments must be `SEXP`.

**Comment by eddelbuettel** [comment]
I got a similar email from Tomas and already fixed it (in `digest`).  Based on that experience, there are two orthogonal issus here:
- Tomas right point out that one should register for `.Call()` when the signature is different (doh)
- One can still register for export by another package, that does not require a SEXP signature (I think...)

Haven't looked at `xts` yet.

**Comment by kalibera** [comment]
Yes, functions for use with `RegisterCCallable`/`GetCCallable` only (not `.Call`) do not have such restrictions.

**Comment by joshuaulrich** [comment]
Thank you both for your input! It looks like I only need to stop registering those functions for `.Call()`. Neither of them are called via `.Call()` in xts:

```bash
josh@thinkpad: ~/git/xts/R (master)
> grep -E "\.Call\(.(isXts|do_merge_xts)" *

```

That also means RcppXts should be okay, since the function signatures do not need to change.

**Comment by joshuaulrich** [comment]
This was fixed in #337.
